bindgen/Context/Function.zig

```diff
@@ -430,6 +437,62 @@ pub const Parameter = struct {
     }
 };
 
+const return_type_init_map: StaticStringMap([]const u8) = .initComptime(.{
+    // builtins
+    .{ "Vector2", ".zero" },
+    .{ "Vector3", ".zero" },
+    .{ "Vector4", ".zero" },
+    .{ "Vector2i", ".zero" },
+    .{ "Vector3i", ".zero" },
+    .{ "Vector4i", ".zero" },
+    .{ "Basis", ".identity" },
+    .{ "Transform2D", ".identity" },
+    .{ "Transform3D", ".identity" },
+    .{ "Projection", ".identity" },
+
+    // primitives
+    .{ "bool", "false" },
+
+    .{ "u8", "0" },
+    .{ "u16", "0" },
+    .{ "u32", "0" },
+    .{ "u64", "0" },
+
+    .{ "i8", "0" },
+    .{ "i16", "0" },
+    .{ "i32", "0" },
+    .{ "i64", "0" },
+
+    .{ "f32", "0.0" },
+    .{ "f64", "0.0" },
+
+    .{ "void", "undefined" },
+});
+
+/// Returns the initializer for a given return type.
+///
+/// Note: Do not use for constructors.
+fn getReturnTypeInitializer(return_type: Type) ?[]const u8 {
+    switch (return_type) {
+        inline .string_name, .string => return ".init()",
+        inline .class, .basic => |type_name| {
+            if (return_type_init_map.get(type_name)) |initializer| {
+                return initializer;
+            }
+
+            if (std.zig.isPrimitive(type_name)) {
+                std.debug.panic("No initializer defined for primitive type '{s}'", .{type_name});
+            }
+
+            // this should exist for all basic and class types
+            return ".init()";
```

Comment:
afaik this won't work since the constructors call out to Godot's APIs, which can't be done statically

Reply:
oh, wait, unless this is only ever used inside a function